### PR TITLE
feat: [SRM-11685]: fix GCO runtime query

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 5 Issues 
* Code issue 5 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11044)

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts [hunk 1]
```diff
@@ -111,6 +111,10 @@ export function transformGCOMetricHealthSourceToGCOMetricSetupSource(sourceData:
       metricTags[tag] = ''
     }
 
+    const parsedQuery =
+      getMultiTypeFromValue(metricDefinition.jsonMetricDefinition as any) === MultiTypeInputType.FIXED
+        ? JSON.stringify(metricDefinition.jsonMetricDefinition, null, 2)
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts:116]: will this always be a valid object?

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx [hunk 2]
```diff
@@ -346,7 +355,15 @@ export function GCOMetricsHealthSource(props: GCOMetricsHealthSourceProps): JSX.
                             formikProps.setFieldError('tooManyMetrics', '')
                           }
                         }}
-                        isQueryExecuted={shouldShowChart}
+                        noDataMessage={
+                          getMultiTypeFromValue(formikProps.values.query) === MultiTypeInputType.FIXED
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx:359]: Please move the logic to utils

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx [hunk 3]
```diff
@@ -346,7 +355,15 @@ export function GCOMetricsHealthSource(props: GCOMetricsHealthSourceProps): JSX.
...
+                            ? getString('cv.monitoringSources.gcoLogs.submitQueryToSeeRecords')
+                            : getString('cv.customHealthSource.chartRuntimeWarning')
+                        }
+                        isQueryExecuted={
+                          shouldShowChart &&
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx:364]: same.

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx]: done

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx]: done

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**